sway/input/cursor.c

```diff
@@ -383,20 +402,21 @@ void dispatch_cursor_axis(struct sway_cursor *cursor,
 static void handle_pointer_axis(struct wl_listener *listener, void *data) {
 	struct sway_cursor *cursor = wl_container_of(listener, cursor, axis);
 	struct wlr_event_pointer_axis *event = data;
-	cursor_handle_activity(cursor, IDLE_SOURCE_POINTER);
+	cursor_handle_activity(cursor, event->device);
 	dispatch_cursor_axis(cursor, event);
 	transaction_commit_dirty();
 }
 
 static void handle_pointer_frame(struct wl_listener *listener, void *data) {
 	struct sway_cursor *cursor = wl_container_of(listener, cursor, frame);
-	cursor_handle_activity(cursor, IDLE_SOURCE_POINTER);
```

Comment:
We don't have a reference to the device here, so I just removed the call. We shouldn't really be seeing frame events without axis/button events, which would already update the idle status.